dotCMS/src/main/java/org/apache/felix/framework/OSGISystem.java

```diff
@@ -0,0 +1,330 @@
+        felixProps.put("org.osgi.framework.startlevel.beginning", "2");
+        felixProps.put("org.osgi.framework.storage.clean", "onFirstInit");
+        felixProps.put("felix.log.level", "3");
+        felixProps.put("felix.fileinstall.disableNio2", "true");
+        felixProps.put("gosh.args", "--noi");
+
+        // Create host activator;
+        final HostActivator hostActivator = HostActivator.instance();
+        felixProps.put(FelixConstants.SYSTEMBUNDLE_ACTIVATORS_PROP, ImmutableList.of(hostActivator));
+
+        return felixProps;
+    }
+
+    /**
+     * Initializes the System framework OSGi using the servlet context
+     *
+     * @return Framework
+     */
+    public synchronized Framework initializeFramework() {
+
+        if(felixFramework != null) {
+
+            return felixFramework;
+        }
+
+        // load all properties and set base directory
+        final Properties felixProps =  defaultProperties();
+
+        try {
+
+            // Set all OSGI Packages
+            String extraPackages;
+            try {
+                extraPackages = getExtraOSGIPackages();
+            } catch (IOException e) {
+                Logger.error(this, "Error loading the OSGI framework properties: " + e);
+                throw new RuntimeException(e);
+            }
+
+            // Setting the OSGI extra packages property
+            felixProps.setProperty(PROPERTY_OSGI_PACKAGES_EXTRA, extraPackages);
+
+            // before init we have to check if any new bundle has been upload
+            // Create an instance and initialize the framework.
+            final FrameworkFactory factory = this.getFrameworkFactory();
+            felixFramework = factory.newFramework(felixProps);
+            felixFramework.init();
+
+            AutoProcessor.process(felixProps, felixFramework.getBundleContext());
+
+            felixFramework.start();
+            Logger.info(this, () -> "Osgi Felix System Framework started");
+        } catch (Exception ex) {
+            felixFramework=null;
+            Logger.error(this, "Could not create OSGI SYSTEM framework: " + ex);
+            throw new RuntimeException(ex);
+        }
+
+        System.setProperty("system"+WebKeys.OSGI_ENABLED, "true");
+
+        return felixFramework;
+    }
+
+    /**
+     * Returns the packages inside the <strong>osgi-extra.conf</strong> and the
+     * osgi-extra-generate.conf files If neither of those files are there, it will generate the
+     * osgi-extra-generate.conf based off the classpath for the OSGI configuration property
+     * <strong>org.osgi.framework.system.packages.extra</strong>. <br/><br/> The property
+     * <strong>org.osgi.framework.system.packages.extra</strong> is use to set the list of packages
+     * the dotCMS context in going to expose to the OSGI context.
+     *
+     * @return String
+     * @throws IOException Any IOException
+     */
+    public String getExtraOSGIPackages() throws IOException {
```

Comment:
Most of these methods should be private, I think.  No reason to expose them publically